src/TesApi.Web/BatchScheduler.cs

```diff
@@ -539,21 +649,48 @@ private async Task<CloudTask> ConvertTesTaskToBatchTaskAsync(TesTask task)
 
             var executorImageIsPublic = (await azureProxy.GetContainerRegistryInfoAsync(executor.Image)) == null;
 
-            var taskCommand = $@"
-                docker pull --quiet {BlobxferImageName} && \
-                {(executorImageIsPublic ? $"docker pull --quiet {executor.Image} && \\" : "")}
-                docker run --rm {volumeMountsOption} --entrypoint=/bin/sh {BlobxferImageName} {downloadFilesScriptPath} && \
-                chmod -R o+rwx /mnt{cromwellPathPrefixWithoutEndSlash} && \
-                docker run --rm {volumeMountsOption} --entrypoint= --workdir / {executor.Image} {executor.Command[0]} -c '{ string.Join(" && ", executor.Command.Skip(1))}' && \
-                docker run --rm {volumeMountsOption} --entrypoint=/bin/sh {BlobxferImageName} {uploadFilesScriptPath}
-            ";
+            var sb = new StringBuilder();
+
+            sb.AppendLine($"write_kv() {{ echo \"$1=$2\" >> /mnt{metricsPath}; }} && \\");
```

Comment:
possible to add some comments to explain some of these? what is "write_kv()" etc? But also what is the script doing overall? short summary will be great. also if order matters.